packages/rest/src/router/routing-table.ts

```diff
@@ -150,6 +160,9 @@ export class RoutingTable {
 
     validateApiPath(route.path);
     this._router.add(route);
+    if (route.path === '/*') {
```

Comment:
@raymondfeng you had some reservations about this.
> This is a bit hacky to test path === '*'. The special route is also added to the underlying router, which might cause unexpected results.

Can you elaborate the concerns, so that we can start a discussion?

Reply:
I am still not able to see what the potential problem is. We run `validateApiPath(route.path)` and `/*` is a valid key to the static assets router. What am I missing?

Reply:
It's not helpful to have `/*` in the router which won't match any requests.

Reply:
+1 on what Raymond said.

>  the static asset route is a special one outside of the Router responsibility. It serves as the last matching candidate.
> I'm fine to have `/*` as the path value, but don't call `this._router.add(route);`.

Let me cross-post my earlier comment https://github.com/strongloop/loopback-next/pull/1848#discussion_r225478536 too:

> IIUC, this is relying on the fact that the code paths invoked here is bypassing path validations and that our routing table will pass the path `*` to Express router without modifications. 
>
> I find that rather brittle, let's find a different solution please.
>
> I am thinking about the following implementation of [`RoutingTable#find`](https://github.com/strongloop/loopback-next/blob/a2ab86d3a3c721f28159585ca4e87bd8504476c4/packages/rest/src/router/routing-table.ts#L173-L188):

```ts
class RoutingTable {
  // ...

  find(request: Request): ResolvedRoute {
    debug('Finding route %s for %s %s', request.method, request.path);

    const found = this._router.find(request);

    if (found) {
      debug('Route matched: %j', found);
      return found;
    }

    debug('No API route found for %s %s, trying to find a static assets', request.method, request.path);
    return this._staticAssetsRoute;
  }
}
```
---

Nitpick: I would personally use just `*` as the path value, or even an empty string, because this is just a placeholder value that should be never used, but this is a detail not worth long discussions - pick whatever makes most sense to you.

Reply:
Got it now.